### ISSUE TITLE
`InfluxDBClient.request` discards error details when raising exception

### ISSUE THREAD

**Issue by thomir** [influxdata/influxdb-python#533]
The `InfluxDBClient.request` method contains code to make a request to the influxdb server that looks like this:

```python
    while retry:
            try:
                response = self._session.request(...)
                break
            except requests.exceptions.ConnectionError as e:
                _try += 1
                if self._retries != 0:
                    retry = _try < self._retries
        else:
            raise requests.exceptions.ConnectionError
```

The problem is that if we fail to connect, the ConnectionError that is raised does not have any of the information about what actually failed, making it really hard to debug from the exception message.

A possibly alternative might be something like this:

```python
    while retry:
            try:
                response = self._session.request(...)
                break
            except requests.exceptions.ConnectionError as e:
                _try += 1
                if self._retries != 0:
                    retry = _try < self._retries
                if not retry:
                    raise
```

This will re-raise the exception that caused the failure, _with_ the contextual information about what actually happened.

**Comment by xginn8** [comment]
i believe this was fixed in https://github.com/influxdata/influxdb-python/pull/508, can you confirm that? you may need to run from HEAD, as this change hasn't made it into a release yet.

**Comment by thomir** [comment]
I can confirm that it's fixed in #508, thanks.
